File: 준영/프로그래머스/LV2/프린터.js
function solution(priorities, location) {
  var answer = 0
  var temp = priorities.map((x, i) => [i, x])
  
  var result_stack = []

  while (temp.length) {
    var temp_x = temp.shift()
    var idx = temp_x[0]
    var x = temp_x[1]
    
    var is_possible = true

    for (const i of temp) {
      if (i[1] > x) {
        is_possible = false
        temp.push([idx, x])
        break
      }
    }

    if (is_possible) {
      result_stack.push(idx)
    }
  }
  answer = result_stack.indexOf(location) + 1
  return answer
}

File: 준영/프로그래머스/LV2/프린터.py
def solution(priorities, location):
    answer = 0
    idx = 0
    temp = []
    for i in priorities:
        temp.append([idx, i])
        idx += 1
        
    result_stack = []

    while temp:
        idx, x = temp.pop(0)
        is_possible = True
        for i in temp:
            if i[1] > x:
                is_possible = False
                temp.append([idx, x])
                break
        
        if is_possible:
            result_stack.append(idx)
    answer = result_stack.index(location)+1
    return answer